setup.py

```diff
@@ -39,6 +30,17 @@
       author_email='info@precice.org',
       license='LGPL-3.0',
       packages=['fenicsprecice'],
-      install_requires=['pyprecice>=3.0.0.0', 'scipy', 'numpy>=1.13.3, <2', 'mpi4py'],
+      setup_requires=['mpi4py<4'],
```

Comment:
I'm not sure what `setup_requires` exactly does but I would not put a hard requirement here for the sake of the check that we have below. If the check does not succeed there is probably a good reason for the warning. We should not try to fix a broken FEniCS installation with this `setup.py`.

Reply:
I looked a bit into the reason why `mpi4py` is failing and actually we are exactly trying to do this: We are fixing a broken FEniCS installation and this is also what we need to do because I think we cannot expect that there will be a corresponding fix for (legacy) FEniCS (either supporting `mpi4py>=4` or restricting to `mpi4py<4`).